enumerables.rb

```diff
@@ -1,37 +1,72 @@
 module Enumerables
+
   def my_each
-    count = 0
-    while count < self.size do
-      yield(self[count])
-      count += 1
+    return to_enum unless block_given?
+
+    for i in 0..self.length
+      yield(self[i])
     end
-    self
   end
```

Comment:
- [ ]  enumerables #my_each does not mutate the original array
     Failure/Error: array.my_each { |num| num + 1 }

- [ ]  enumerables #my_each when a block is given when self is an array returns the array itself after calling the given block once for each element in self
     Failure/Error: let(:block) { proc { |num| num < (LOWEST_VALUE + HIGHEST_VALUE) / 2 } }
- [ ] enumerables #my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self
     Failure/Error: expect(range.my_each(&block)).to eq(range.each(&block))

- [ ] enumerables #my_each when a block is given when self is a range behaves like #each calls the given block once for each element in self
     Failure/Error: enum.my_each(&block)

- [ ] enumerables #my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self
     Failure/Error: expect(my_each_output).to eq(each_output)
     
       expected: "[:a, 1][:b, 2][:c, 3][:d, 4][:e, 5]"
            got: ""